EIPS/eip-5528.md

```diff
@@ -0,0 +1,183 @@
+ - etc
+
+The buyer-contract and seller-contract should not have constraint rules.
+
+Let's discuss the following functions.
+
+1. **constructor**
+
+An escrow contract will define success/failure conditions. It means constraint rules might not be changed forever (might be changed after being created for the market exchange rate.), so it guarantees escrow policy.
+
+2. **escrowFund**
+
+This function should run differently for buyers and sellers.
+
+2.1 [Seller]
+- The seller calls this function to be escrow-ready. The seller's token ownership(balance) will be transferred to the escrow contract and the escrow balance should be `(Seller: amount, Buyer: 0)`.
+- The seller can call this function multiple times depending on implementation, but preferred just one time.
+
+2.2 [Buyer]
+- During the escrow process, the buyer should be able to call this function to deposit funds into the escrow account.
+- The escrow balance should be  `(Seller: amount X exchange rate, Buyer: amount)`. The Buyer: the amount will be used for the refund process.
+- In a successful scenario, the seller's escrow balance should be `(Seller: -= amount X exchange rate, Buyer: += amount)`.
+
+3. **escrowRefund**
+
+This function should be invoked by buyers only.
+The buyer can call this function in the running state only. In the state of failure or success, could not be a success.
+The escrow balances of seller and buyer will be updated reverse way of `escrowFund`
+
+
+4. **escrowWithdraw**
+
+Buyers and sellers can withdraw tokens from the escrow account to their account.
+The following processes are recommended.
+- Buyer can withdraw in escrow-success state only. Ownership of seller tokens can be transferred to the buyer from escrow-contract. In an escrow-failed state, the buyer should call the `escrowRefund` function.
+- When the seller calls this function in the escrow-success state, the remaining seller token will be transferred to the seller, and the earned buyer's token will be also transferred from the escrow-account.
+- In the case of escrow-failed, the seller only gets a refund seller token.
+
+## Backwards Compatibility
+
+This EIP is fully backward compatible with the [EIP-20](./eip-20.md) specification.
+
+
+## Test Cases
+
+1. [Seller/Buyer Token example](../assets/eip-5528/ERC20Mockup.sol).
+2. [Escrow contract example](../assets/eip-5528/EscrowContractAccount.sol).
+3. [Unit test example with truffle](../assets/eip-5528/truffule-test.js).
+
+The above 3 files demonstrate the following conditions for exchanging seller/buyer tokens.
+- The exchange rate is one to one.
+- If the number of buyers reaches 2, the escrow process will be terminated(success).
+- Otherwise(not meet success condition yet), buyers can refund(or withdraw) their funded tokens.
+
+## Security Considerations
+
+Since the external contract(Escrow Contract) will control seller or buyer rights, flaws within the escrow contract directly lead to the standard’s unexpected behavior.
```

Comment:
```suggestion
Since the escrow contract will control seller and buyer rights, flaws within the escrow contract will directly lead to unexpected behavior and potential loss of funds.
```
Recommended: Rephrase